mirror: T3169: Add smoke testing and fix dependency issues

jack9603301 opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Text placed between these delimiters is considered a commend and is not rendered -->

## Change Summary
<!--- Provide a general summary of your changes in the Title above -->
Add smoke testing and fix dependency issues


## Types of changes
<!--- What types of changes does your code introduce? Put an 'x' in all the boxes that apply. -->
<!--- NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking the box, please use [x] --> 
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- All submitted PRs must be linked to a Task on Phabricator. -->
https://phabricator.vyos.net/T3169

## Component(s) name
<!-- A rather incomplete list of components: ethernet, wireguard, bgp, mpls, ldp, l2tp, dhcp ... -->
mirror

## Proposed changes
<!--- Describe your changes in detail -->
Add smoke testing and fix dependency issues

## How to test
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

jack9603301 commented:
Smoke test passed in local environment

c-po commented on smoketest/scripts/cli/test_interfaces_bridge.py:
```diff
@@ -51,14 +51,16 @@ def setUp(self):
     def tearDown(self):
-        self.session.delete(['interfaces','bonding'])
-        self.session.delete(['interfaces','dummy'])
+        self.session.delete(['interfaces','bonding','bond1'])
```
Why is this change required?

jack9603301 replied:
In fact, it doesn't need to be modified here. However, in order to ensure that the redundant interface will not be deleted, it is improved by the way. Do you want to restore the original deletion instruction?

jack9603301 replied:
In fact, there is no need to change it here. But in order to ensure that the redundant interface is not deleted, it is improved

c-po commented on smoketest/scripts/cli/test_interfaces_ethernet.py:
```diff
@@ -66,6 +66,10 @@ def setUp(self):
             self._macs[interface] = mac
 
+        # Creating test interfaces for port mirroring
+        self.session.set(['interfaces','dummy','dum100'])
```
Please use spaces for list items.

c-po commented on smoketest/scripts/cli/test_interfaces_bonding.py:
```diff
@@ -50,6 +50,14 @@ def setUp(self):
             self._options['bond0'].append(f'member interface {member}')
 
+        # Creating test interfaces for port mirroring
+        self.session.set(['interfaces','dummy','dum10'])
```
Use spaces for list items.

jack9603301 committed:
smoketest: mirror: T3169: Re implement smoke test of span (mirror)

In the submission of ef62950, the smoke test was deleted
 for various reasons. Now, it's time to re implement it

jack9603301 committed:
mirror: T3169: Fix SPAN dependency issues

Fix that when the monitored interface does not exist,
 the span function will cause an error because the monitored interface does not exist

c-po commented on smoketest/scripts/cli/test_interfaces_bonding.py:
```diff
@@ -50,6 +50,14 @@ def setUp(self):
         for member in self._members:
             self._options['bond0'].append(f'member interface {member}')
 
+        # Creating test interfaces for port mirroring
```
Speciy `dum10` as value and probably use a dum interface number > 10000

c-po commented on smoketest/scripts/cli/test_interfaces_bridge.py:
```diff
@@ -51,6 +51,7 @@ def setUp(self):
             self.session.set(['interfaces','bonding','bond1','member','interface','dum0'])
             self.session.set(['interfaces','bonding','bond1','member','interface','dum1'])
             self._members.append('bond1')
+            self._mirror_interfaces.append('bond1')
```
This should not be done for ALL testscases, rather limit this to the specific test.

c-po commented on smoketest/scripts/cli/test_interfaces_bonding.py:
```diff
@@ -50,6 +51,15 @@ def setUp(self):
+                self.session.set(['interfaces', 'dummy', mon_intf])
+    
+    def tearDown(self):
+        self.session.delete(['interfaces','dummy','dum10'])
```
Why `dum10`, you really need to be more careful!

jack9603301 replied:
Oh, I haven't tested it yet. I'm just going to 'push' it first, and I'm preparing for the test

c-po commented on smoketest/scripts/cli/test_interfaces_bridge.py:
```diff
@@ -35,6 +35,7 @@ def setUp(self):
 
         self._base_path = ['interfaces', 'bridge']
         self._interfaces = ['br0']
+        self._mirror_interfaces = ['bond1']
```
Who creates bond1?

jack9603301 replied:
The code for creating `bond1` is below. I share `bond1`

jack9603301 committed:
smoketests: mirror: T3169: Improve test dependent interface creation

jack9603301 committed:
smoketests: mirror: T3169: Modify the interface number to more than 10000

jack9603301 commented:
The interface number is now greater than 10000

c-po commented on smoketest/scripts/cli/test_interfaces_bridge.py:
```diff
@@ -37,8 +37,10 @@ def setUp(self):
 
         self._base_path = ['interfaces', 'bridge']
         self._interfaces = ['br0']
+        
```
Why do we need a blank line?

c-po commented on smoketest/scripts/cli/test_interfaces_bridge.py:
```diff
@@ -48,9 +50,22 @@ def setUp(self):
                 if not '.' in tmp:
                     self._members.append(tmp)
 
+            self.session.set(['interfaces','dummy','dum10001'])
```
You continously do not follow the code path. I requested to move this out of the general interface setup for multiple times now.
